### PR TITLE
Enabling metadata upload in react refactor (SCP-3667)

### PR DESCRIPTION
**Pull request by devonbush**
SCP-3667   More-or-less a direct UX port of the existing metadata upload page

image

TO TEST:
1. make sure react-upload-wizard feature flag is on
2. go to upload wizard, and select Metadata step
3. confirm page more-or-less matches current production UX
4. toggle the 'convention_required' feature flag
5. confirm UX works to upload and parse a file

### CODE REVIEW

File: app/javascript/components/upload/MetadataStep.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    Why use the metadata convention?<br/>
+    Using the metadata convention will enable your data to be discovered through our faceted search interface.
+    Using the convention means extra validation will be applied to ensure your metadata conforms to standard vocabularies.<br/>
+    <a href='https://singlecell.zendesk.com/hc/en-us/articles/360061006411-Metadata-Convention' target='_blank' rel="noopener noreferrer">Learn more</a>.
+  </Popover>
+)
+
+const conventionIssuePopover = (
+  <Popover id="convention-issues">
+    Using the convention is now required.  If this presents a particular issue for your study,
```
Comment (bistline) [inline app/javascript/components/upload/MetadataStep.js:136]: We should update this with the new UX from #1178.

File: app/javascript/components/upload/UploadWizard.js [hunk 2]
```diff
@@ -179,43 +181,45 @@ export default function UploadWizard({ studyAccession, name }) {
...
-      <div className="col-md-10">
-        <h4>{studyAccession}: {name}</h4>
+  return <UserProvider>
+    <div className="">
+      <div className="row padded">
+        <div className="col-md-10">
+          <h4>{studyAccession}: {name}</h4>
+        </div>
+        <div className="col-md-2">
+          <a href={`/single_cell/study/${studyAccession}`}>View Study</a>
```
Comment (bistline) [inline app/javascript/components/upload/UploadWizard.js:191]: This link could use styling as a Bootstrap button - it is very easy to miss without it.

File: app/javascript/components/upload/MetadataStep.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  component: MetadataForm,
+  fileFilter: metadataFileFilter
+}
+
+/** Renders a form for uploading one or more cluster/spatial files */
+function MetadataForm({
+  formState,
+  addNewFile,
+  updateFile,
+  saveFile,
```
Comment (bistline) [inline app/javascript/components/upload/MetadataStep.js:30]: This works as advertised, but I feel like there is a UX regression here in terms of a successful upload/save.  There is no visual indication to the user that the action succeeded.  I'm not necessarily advocating for a modal (we have far too many, and they are very obtrusive), but the current UX (non-React) is _very_ explicit when things work, and when they don't.  I'd like to find a middle ground that gives necessary feedback to the user without taking over the page.

**Review by bistline — CHANGES_REQUESTED** [review]
Metadata form works as advertised, but I think the lack of feedback to users on successful upload should be addressed sooner rather than later.  I thought that there was an error at first because it looked like nothing happened after the upload finished, and I only realized it worked by looking at the logs.

File: app/javascript/components/upload/UploadWizard.js (hunk 2 above)
Comment (devonbush) [reply app/javascript/components/upload/UploadWizard.js:191]: This is addressed in the later (but now-merged) UX refinement PR

File: app/javascript/components/upload/MetadataStep.js (hunk 3 above)
Comment (devonbush) [reply app/javascript/components/upload/MetadataStep.js:30]: yup, reasonable success/fail/parsing alerts haven't been done yet -- Jerome is still brainstorming concepts.  I'll create a separate ticket to make sure it gets addressed

File: app/javascript/components/upload/MetadataStep.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+import { UserContext } from 'providers/UserProvider'
+import FileUploadControl, { FileTypeExtensions } from './FileUploadControl'
+import { TextFormField, SavingOverlay, SaveDeleteButtons } from './uploadUtils'
+
+const DEFAULT_NEW_METADATA_FILE = {
+  file_type: 'Metadata',
+  use_metadata_convention: true
+}
+
+const metadataFileFilter = file => file.file_type === 'Metadata'
```
Comment (ehanna4) [inline app/javascript/components/upload/MetadataStep.js:16]: Does this need a check if file exists? 
ie `file?.file_type`

File: app/javascript/components/upload/MetadataStep.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+      </div>
+
+    </div>
+    { file &&
+      <div className="row top-margin" key={file._id}>
+        <div className="col-md-12">
+          <form id={`metadataForm-${file._id}`}
+            className="form-terra"
+            acceptCharset="UTF-8"
+            onSubmit={() => { console.log('submitting'); return false}}>
```
Comment (ehanna4) [inline app/javascript/components/upload/MetadataStep.js:70]: Should this` console.log` be removed?

File: app/javascript/components/upload/MetadataStep.js [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    Using the convention means extra validation will be applied to ensure your metadata conforms to standard vocabularies.<br/>
+    <a href='https://singlecell.zendesk.com/hc/en-us/articles/360061006411-Metadata-Convention' target='_blank' rel="noopener noreferrer">Learn more</a>.
+  </Popover>
+)
+
+const conventionIssuePopover = (
+  <Popover id="convention-issues">
+    Using the convention is now required.  If this presents a particular issue for your study,
+    please contact us at <a href='mailto:scp-support@broadinstitute.zendesk.com'>scp-support@broadinstitute.zendesk.com</a>
+    so we can help or make any updates to the convention.
```
Comment (ehanna4) [inline app/javascript/components/upload/MetadataStep.js:138]: need a space between the link and 'so'

**Comment by ehanna4** [comment]
Unsure if this is intended (maybe since the metadata file is still uploading?) but the cluster files are normal while the metadata file name is italicized 
Screen Shot 2021-09-26 at 9 29 28 PM

**Review by ehanna4 — APPROVED** [review]
Overall looks good! Tested clicking around the UX and uploading a file. Was a "bad file" from the test data in SCP-core but I'm figuring the file validation is not included in this PR? (Looks like more convo in other comments about this success/fail upload being addressed in a different ticket)

Left a couple of comments but otherwise looks good 👍

**Comment by devonbush** [comment]
Now updated with the zendesk form UX 
image

**Comment by devonbush** [comment]
> Unsure if this is intended (maybe since the metadata file is still uploading?) but the cluster files are normal while the metadata file name is italicized
> Screen Shot 2021-09-26 at 9 29 28 PM

I think this is probably because you were using a 'bad' file, and so it would get deleted from the server.  In any case, we have a separate ticket to clean up that UX for showing why things are or are not italicized

**Pull request merged**